subsys/bluetooth/mesh/shell/rpr.c

```diff
@@ -38,9 +38,12 @@ static void rpr_scan_report(struct bt_mesh_rpr_cli *cli,
 		uint8_t len, type;
 		uint8_t data[31];
 
-		len = net_buf_simple_pull_u8(adv_data) - 1;
 		type = net_buf_simple_pull_u8(adv_data);
-		memcpy(data, net_buf_simple_pull_mem(adv_data, len), len);
+		len = net_buf_simple_pull_u8(adv_data);
+		if (len > 0) {
+			len = MIN(len, sizeof(data) - 1);
+			memcpy(data, net_buf_simple_pull_mem(adv_data, len), len);
```

Comment:
ouch, the assumptions was that `len = net_buf_simple_pull_u8(adv_data);` would return a valid `len`

Reply:
If that assumption was always true, we wouldn't need to modify anything here :D

Reply:
That's not correct, the problem could be because `len > sizeof(data[31])` in `memcpy(data, net_buf_simple_pull_mem(adv_data, len), len);`

My assumption here was that this snippet was valid
```
len = net_buf_simple_pull_u8(adv_data);
net_buf_simple_pull_mem(adv_data, len);
```

Reply:
One more thing, in @PavelVPV proposal we get all data from the AD structure, not limiting to `sizeof(data)`, though that is the limit to copy.  That seems the right approach because it advances the right amount of bytes in the buffer.

Reply:
@ceolin as per the Mesh spec, a `len > sizeof(data[31])` would also indicate an invalid len field, as no mesh packets can be longer than that (thuse also why `data[31]` is only 31 octets. I would assume that the Mesh stack verifies the length field before sending it to the shell here, so it's only if the stack introduced a bug that an issue would ever occur here.

Reply:
This is a scan report with AD Structures requested by the Remote Provisioning client. Here it can be any AD Structure associated with the mesh device. And here the mesh spec doesn't put any limitation on the length, but refers to the Bluetooth Core spec (section 4.3.4.9, of MshPRTd1.1r20):
> If present, the AdvStructures field contains a concatenated list of AD Structures with information requested by the Remote Provisioning Client. The value has the same format as advertising data or scan response data, as defined in [1] Vol 3, Part C, Section 11. 

And Vol 3, Part C, Section 11 of the Bluetooth Core doesn't put any limitation on the length of AD Structure's data.